go/private/rules/cross.bzl

```diff
@@ -0,0 +1,74 @@
+# Copyright 2014 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("//go/private/rules:transition.bzl", "go_cross_transition")
+load("//go/private:providers.bzl",
+     "GoLibrary",
+     "GoSource",
+     "GoArchive",
+)
+
+def _go_cross_impl(ctx):
+    old_default_info = ctx.attr.target[DefaultInfo]
+    old_executable = old_default_info.files_to_run.executable
+    if not old_executable:
+        fail('target must have executable')
```

Comment:
Do you have ideas on how switch to between a executable and non executable version of `go_cross`. I see how `go_binary` does it in wrappers.bzl. However, it seems to me like there’s no way for a `go_cross_wrapper` macro to access the attributes of the `go_binary` specified by the `target` attribute. I could use `native.existing_rule` but as far as I can tell that would only work if the rule the `target` Label refers to is in the same package as the `go_cross` rule.

Reply:
I updated the diff to have a `go_cross` wrapper macro that tries to find the target in the local package, but if it fails it suggests that the user explicitly call `go_cross_executable` or `go_cross_non_executable`. This feels a little gross to me, so let me know if you have a better suggestion.

Reply:
Thanks for trying this out. It also feels rather invasive to me, so I would prefer a different solution.

Simple idea first: How common is it for `go_cross` targets to be `bazel run`? With `platform` set, it would be relatively unlikely that the target platform matches host.

Next simplest idea: Unconditionally mark the rule as executable, but write and advertise a bash/bat script or simple Go binary that prints an error message when executed in case the link mode doesn't provide you with an actual `executable`.

Reply:
Since one can specify a `go_cross` rule without a platform and with just an `sdk_version`, I decided to go for the second option. This way users can still `bazel run` a `go_cross` rule that just changes the sdk version.